tests/synced_collection_test.py

```diff
@@ -531,9 +531,7 @@ def test_set_get_numpy_float_data(self, synced_collection, dtype, shape):
         )
 
         if has_corresponding_python_type:
-            with pytest.raises((ValueError, TypeError)), pytest.warns(
-                NumpyConversionWarning
-            ):
+            with pytest.raises((ValueError, TypeError)):
```

Comment:
We had some code paths that were using both `pytest.raises` and `pytest.warns` context managers, but the warning was never issued because the error was raised first. Pytest basically ignored that the warning wasn't being raised, but it changed its behavior here in version 8 to correctly expect the warning, which was never issued before. https://github.com/pytest-dev/pytest/issues/9036